mlflow/models/evaluation/base.py

```diff
@@ -958,6 +951,24 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
     raise ModelValidationFailedException(message=os.linesep.join(failure_messages))
 
 
+def _convert_data_to_mlflow_dataset(data, targets=None):
+    """Convert input data to mlflow dataset."""
+    if "pyspark" in sys.modules:
+        from pyspark.sql import DataFrame as SparkDataFrame
+
+    if isinstance(data, list):
+        return mlflow.data.from_numpy(np.array(data), targets=np.array(targets))
+    elif isinstance(data, np.ndarray):
+        return mlflow.data.from_numpy(data, targets=targets)
+    elif isinstance(data, pd.DataFrame):
+        return mlflow.data.from_pandas(df=data, targets=targets)
+    elif "pyspark" in sys.modules and isinstance(data, SparkDataFrame):
+        return mlflow.data.from_spark(df=data, targets=targets)
+    else:
+        # Cannot convert to mlflow dataset, return original data.
+        return data
```

Comment:
nit: can we emit an info/warning message that the dataset could not be converted? users may expect their eval dataset to be logged, so we should let them know if it will not be.